src/main/java/ch/uzh/ifi/hase/soprafs22/repository/GameRepository.java

```diff
@@ -12,14 +12,26 @@
 public interface GameRepository extends JpaRepository<Game, Long> {
     Game findById(long id);
 
-    @Query("select g from Game g where g.gameStatus = ch.uzh.ifi.hase.soprafs22.constant.GameStatus.ACTIVE " +
-            "and g.userId <> :userId " +
-            "and g not in (select g from Game g join Play p on g.id = p.gameId where p.userId = :userId)")
+
+    //Has to check both directions because user only wants to see cards from people that user is interested in and that are interested in user
+    @Query("select game from Game game where game.gameStatus = ch.uzh.ifi.hase.soprafs22.constant.GameStatus.ACTIVE " +
+            "and game.userId <> :userId " +
+            "and game not in (select g from Game g join Play p on g.id = p.gameId where p.userId = :userId)" +
+            "and game not in (select g from Game g join User u on g.userId=u.id where u.age < (select u.minAge from User u where u.id = :userId))" +
+            "and game not in (select g from Game g join User u on g.userId=u.id where u.age > (select u.maxAge from User u where u.id = :userId))" +
+            "and game not in (select g from Game g join User u on g.userId=u.id where u.minAge > (select u.age from User u where u.id = :userId))"+
+            "and game not in (select g from Game g join User u on g.userId=u.id where u.maxAge < (select u.age from User u where u.id = :userId))" +
+            "and game in(select g from Game g, User u1, User u2 where u1.id =:userId and u2.gender member of u1.genderPreferences and u1.gender member of u2.genderPreferences and u2.id = g.userId)")
     Page<Game> getOtherUserWithActiveGameThatWasNotPlayedOn(Pageable pageable, @Param("userId") long userId);
 
     @Query("select count (g) from Game g where g.gameStatus = ch.uzh.ifi.hase.soprafs22.constant.GameStatus.ACTIVE " +
             "and g.userId <> :userId " +
-            "and g not in (select g from Game g join Play p on g.id = p.gameId where p.userId = :userId)")
+            "and g not in (select g from Game g join Play p on g.id = p.gameId where p.userId = :userId)"+
+            "and g not in (select g from Game g join User u on g.userId=u.id where u.age < (select u.minAge from User u where u.id = :userId))" +
```

Comment:
I imagine this query to be expensive anyway, So maybe it would be better to just pass minAge, maxAge, and genderPreferences as parameters instead of nest another query into the already nested query.

Reply:
The query seems to perform fine and it is given that we can assume to have a small userbase, so i think this is sensible.